### PR TITLE
Hashtagify transformation

### PR DESCRIPTION
**Pull request by pithysr**
This transformation adapts an input sentence by identifying named entities and other common words and turning them into hashtags, as often used in social media.

### CODE REVIEW

File: transformations/hashtagify/transformation.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        perturbed_texts.append(transformed_sentence)
+
+    return perturbed_texts
+
+
+class HashtagifyTransformation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
```
Comment (ashish3586) [inline transformations/hashtagify/transformation.py:91]: TEXT_TAGGING will not be applicable here due to the difference in the number of tokens between the input sentence and the transformed sentence.(New Delhi --> #NewDelhi)

Comment (pithysr) [reply transformations/hashtagify/transformation.py:91]: I agree. It can be removed.

Comment (pithysr) [reply transformations/hashtagify/transformation.py:91]: Should I remove it and commit again?

File: transformations/hashtagify/transformation.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+from nltk.tree import Tree
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+nltkdl("words")
+nltkdl("maxent_ne_chunker")
+nltkdl("punkt")
+nltkdl("averaged_perceptron_tagger")
+nltkdl("stopwords")
```
Comment (kaustubhdhole) [inline transformations/hashtagify/transformation.py:20]: You might want to move this to the constructor.

File: transformations/hashtagify/transformation.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        "visual",
+        "highly-meaning-preserving",
+        "high-precision",
+        "high-coverage",
+        "social-reasoning",
+    ]
+
+    def __init__(self, seed=666, max_outputs=1):
+        super().__init__(seed, max_outputs=max_outputs)
+        self.nlp = spacy.load("en_core_web_sm")
```
Comment (kaustubhdhole) [inline transformations/hashtagify/transformation.py:108]: You can use spacy like [this](https://github.com/GEM-benchmark/NL-Augmenter/blob/d39f0e5f57a79937296351996617ac03ee80fb74/transformations/close_homophones_swap/transformation.py#L55)

File: transformations/hashtagify/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+New Delhi is among the many famous places in India.
+```
+
+to
+
+```
+#NewDelhi is among the many famous places in India.
+```
+
+
```
Comment (kaustubhdhole) [inline transformations/hashtagify/README.md:26]: I would suggest to add the robustness evaluation section too.

File: transformations/hashtagify/transformation.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                        "#" + re.sub(" ", "", cand.title()),
+                        transformed_sentence,
+                    )
+
+        perturbed_texts.append(transformed_sentence)
+
+    return perturbed_texts
+
+
+class HashtagifyTransformation(SentenceOperation):
```
Comment (ashish3586) [inline transformations/hashtagify/transformation.py:87]: Please add docstring for the class HashtagifyTransformation.

**Review by mille-s — APPROVED** [review]

**Comment by kaustubhdhole** [comment]
Hi @pithysr, these changes look great! I would strongly recommend you to do the robustness evaluation too for your PR like [other PRs](https://github.com/GEM-benchmark/NL-Augmenter/tree/main/evaluation). I am merging this now. I think the failed build has been fixed in another PR (not an issue with your PR).

**Review by kaustubhdhole — APPROVED** [review]

**Pull request merged**